### PR TITLE
[Performance]: Process affinity to CPU cores with multiple sockets support

### PR DESCRIPTION
**Pull request by HaiShaw**
## Motivation

Setting process affinity to designated cpu ids, to avoid unwanted process migrations (cost high in case of multiple sockets)

## Modifications

Assign specific process to designated cores, handle multiple sockets, and HT (hyper threading: 1 Physical core runs as 2 logical cores)

Online benchmarking shows notable improvement (request rate: 4/8/16) on MI300X system with dual socket CPUs.

## Checklist

- [+] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/managers/scheduler.py [hunk 1]
```diff
@@ -1398,6 +1400,34 @@ def run_scheduler_process(
     dp_rank: Optional[int],
     pipe_writer,
 ):
+    # current process
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]: wrap this as a unity function in `srt/utils.py`

**Review by merrymercy — CHANGES_REQUESTED** [review]
We have multiple threads, not sure whether we need to call them all.
also call it here https://github.com/sgl-project/sglang/blob/55842eb81a782da7e522ec0210c3fa1f3f74dc0a/python/sglang/srt/managers/tp_worker_overlap_thread.py#L90

**Comment by HaiShaw** [comment]
> We have multiple threads, not sure whether we need to call them all. also call it here
> 
> https://github.com/sgl-project/sglang/blob/55842eb81a782da7e522ec0210c3fa1f3f74dc0a/python/sglang/srt/managers/tp_worker_overlap_thread.py#L90

We don't need to call this per thread, currently we look forward to binding at process level. :)

File: python/sglang/srt/managers/scheduler.py (hunk 1 above)
Comment (HaiShaw) [reply python/sglang/srt/managers/scheduler.py]: Done as suggestion.

File: python/sglang/srt/managers/scheduler.py [hunk 2]
```diff
@@ -1398,6 +1399,9 @@ def run_scheduler_process(
     dp_rank: Optional[int],
     pipe_writer,
 ):
+    # set cpu affinity to this gpu process
+    gpu_proc_affinity(server_args, gpu_id)
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]: Do not pass in the big server_args. Only pass in what is really needed.

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/managers/scheduler.py (hunk 2 above)
Comment (HaiShaw) [reply python/sglang/srt/managers/scheduler.py]: Good point, just revised.

**Pull request merged**